app/src/main/java/org/groundplatform/android/ui/map/CameraUpdateRequest.kt

```diff
@@ -39,7 +41,15 @@ data class NewCameraPositionViaCoordinatesAndZoomLevel(
   val shouldAnimate: Boolean = false,
 ) : CameraUpdateRequest() {
 
-  /** Returns the resolved zoom level based on request parameters. */
-  fun getZoomLevel(currentZoomLevel: Float) =
-    if (isAllowZoomOut) zoomLevel else max(zoomLevel, currentZoomLevel)
+  /**
+   * Determines the effective zoom level to use when centering or restoring the map view.
+   *
+   * Ensures we never zoom in or out beyond a safe range [2f, 21f]. When [isAllowZoomOut] is false,
+   * prevents accidental zoom-out by keeping at least the current zoom level (avoids sudden
+   * jump-outs when reloading or switching maps).
+   */
+  fun getZoomLevel(currentZoomLevel: Float): Float {
+    val target = if (isAllowZoomOut) zoomLevel else max(zoomLevel, currentZoomLevel)
+    return target.coerceIn(DEFAULT_MAP_MIN_ZOOM, DEFAULT_MAP_MAX_ZOOM)
```

Comment:
What was the rationale for locking the zoom level? Can we revert this if it is by mistake?